src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -86,6 +86,11 @@ public MSBuildProjectLoader(Workspace workspace, ImmutableDictionary<string, str
         /// </summary>
         public bool SkipUnrecognizedProjects { get; set; } = true;
 
+        /// <summary>
+        /// 
+        /// </summary>
+        public bool ProduceBinaryLog { get; set; } = false;
```

Comment:
How can I control the name of the binary log file that is produced? Basically how from the perspective of the provided API can I construct a path from which I can read back the binary log?

Reply:
Another way of stating it. Assume I'm a customer that uses this API and sets the value to `true`. How do I write by `File.Open` command to grab the produced binary log off of disk and upload to Azure, etc ...

Reply:
I'd lean toward taking a `string? binaryLogPath`, maybe?

Reply:
Or even simpler, why not just let you pass the ILogger directly? Then then caller can customize to their heart's content.

Reply:
(That lets you specify what logging level they want, or if they want a different logger format entirely, or also a way the tool could maintain it's own internal log if it needed to analyze it directly...)